src/net_processing.cpp

```diff
@@ -1815,9 +1853,9 @@ bool PeerManagerImpl::AlreadyHaveTx(const GenTxid& gtxid)
     return m_recent_rejects.contains(hash) || m_mempool.exists(gtxid);
 }
 
-bool PeerManagerImpl::AlreadyHaveBlock(const uint256& block_hash)
+bool PeerManagerImpl::AlreadyHaveBlock(const CNode& pfrom, const uint256& block_hash)
 {
-    return m_chainman.m_blockman.LookupBlockIndex(block_hash) != nullptr;
+    return LookupBlockIndexForPeer(pfrom, block_hash) != nullptr;
```

Comment:
Should we just get rid of this function and inline this single line in the only place that `AlreadyHaveBlock()` is called?

Reply:
Inlined the call to `LookupBlockIndexForPeer`.